src/state.cairo

```diff
@@ -13,23 +19,56 @@ pub struct ChainState {
     pub epoch_start_time: u32,
     /// Previous timestamps.
     pub prev_timestamps: Span<u32>,
-    // TODO: utreexo_roots?
+    /// Utreexo state
+    pub utreexo_state: @UtreexoState,
+}
+
+/// Accumulator representation of the state aka "Compact State Node"
+#[derive(Drop, Copy)]
```

Comment:
Maybe all utreexo related structures deserve a separate file?

Reply:
Actually, the definitions are pretty compact right now plus there will be some cross dependencies (chain state -> utreexo state, utreexo output -> tx output). So maybe postpone for the future refactoring?